src/LaunchDarkly.EventSource/ConfigurationBuilder.cs

```diff
@@ -138,101 +144,124 @@ public ConfigurationBuilder ReadTimeout(TimeSpan readTimeout)
-        /// </summary>
-        /// <param name="name">the header name</param>
-        /// <param name="value">the header value </param>
-        /// <returns>the builder</returns>
-        public ConfigurationBuilder RequestHeader(string name, string value)
-        {
-            _requestHeaders[name] = value;
-            return this;
-        }
-
+        public ConfigurationBuilder RequestHeaders(IDictionary<string, string> headers)
+        {
+            _requestHeaders = headers ?? throw new ArgumentNullException(nameof(headers));
+            return this;
+        }
+        
+        /// <summary>
+        /// Adds a request header to be sent with each EventSource HTTP request.
+        /// </summary>
+        /// <param name="name">the header name</param>
+        /// <param name="value">the header value </param>
+        /// <returns>the builder</returns>
+        public ConfigurationBuilder RequestHeader(string name, string value)
+        {
+            _requestHeaders[name] = value;
+            return this;
+        }
+
         /// <summary>
         /// Sets the HttpMessageHandler that will be used for the HTTP client, or null for the default handler.
         /// </summary>
-        /// <param name="handler">the message handler implementation</param>
-        /// <returns>the builder</returns>
-        public ConfigurationBuilder MessageHandler(HttpMessageHandler handler)
-        {
-            this._messageHandler = handler;
-            return this;
-        }
-
-        /// <summary>
-        /// Sets the HTTP method that will be used when connecting to the EventSource API.
+        /// <remarks>
+        /// Do not use this property if you are using <see cref="HttpClient"/> to specify a custom HTTP client. Doing so will cause <see cref="Build"/> to throw an exception.
+        /// </remarks>
+        /// <param name="handler">the message handler implementation</param>
+        /// <returns>the builder</returns>
+        public ConfigurationBuilder MessageHandler(HttpMessageHandler handler)
+        {
+            this._messageHandler = handler;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies that EventSource should use a specific HttpClient instance for HTTP requests.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Normally, EventSource creates its own HttpClient and disposes of it when you dispose of the EventSource. If you provide your own HttpClient using this method, you are responsible for managing the HttpClient's lifecycle-- EventSource will not dispose of it.
+        /// </para>
+        /// <para>
+        /// EventSource will not modify this client's properties, so you should not the ConfigurationBuilder methods <see cref="MessageHandler"/> and <see cref="ConnectionTimeout"/> if you use this option. Doing so will cause <see cref="Build"/> to throw an exception.
+        /// </para>
```

Comment:
Rewote this doc comment from the contributed PR, for clarity & to match our style.